Services/Masterdata/Services.Masterdata.EF/Repositories/MasterdataRepository.cs

```diff
@@ -65,6 +67,30 @@ public async Task<ResultPagedListModel<MasterdataModel>> GetMasterdata(string? m
 
         return result;
     }
+
+    public async Task<ResultPagedListModel<string>> GetTags(string masterdataType, QueryModel? querymodel = null)
+    {
+        var masterdataTypeFilter = string.IsNullOrEmpty(masterdataType)
+                    ? "1 = 1"
+                    : Guid.TryParse(masterdataType, out _)
+                        ? $"MasterdataTypeId"
+                        : $"MasterdataTypes.Code";
+
+        var sql = @$"SELECT DISTINCT Tags.value AS Tag
```

Comment:
Raw sql in code is a bad practise. Statement could be converted to linq expressions. Besides this there is a real security issue here, because the masterdatatype isn't treaded as a sql param and creates the possibility of a SQL injection

Reply:
I agree, there shouldn't be raw sql in code. Either convert to linq or a stored procedure, supplying `masterDataType` as a `SqlParameter`.

Reply:
I think it's an acceptable solution. It's fairly straight forward, can't do this in linq (using cross apply and json). Also, there is a bit of a dynamic part when building up the where-statement, that would require some tricky stored procedure. More of a hassle then what it's worth I think. Using a SqlParameter is a good improvement. Changed that.